src/internal/repository-metadata-files/provider/build.go

```diff
@@ -2,74 +2,125 @@ package provider
 
 		var targets = make([]provider.Target, 0)
-		for _, asset := range r.ReleaseAssets.Nodes {
-			if platform := github.ExtractPlatformFromFilename(asset.Name); platform != nil {
-				targets = append(targets, provider.Target{
-					OS:          platform.OS,
-					Arch:        platform.Arch,
-					Filename:    asset.Name,
-					DownloadURL: asset.DownloadURL,
-				})
-			} else if strings.HasSuffix(asset.Name, "SHA256SUMS") {
-				shaSumsArtifact = asset
-			} else if strings.HasSuffix(asset.Name, "SHA256SUMS.sig") {
-				shaSumsSignatureArtifact = asset
-			}
+		for _, a := range versionArtifacts.TargetArtifacts {
+			targets = append(targets, provider.Target{
+				OS:          a.OS,
+				Arch:        a.Arch,
+				Filename:    a.Name,
+				DownloadURL: a.DownloadURL,
+			})
 		}
 		if len(targets) == 0 {
 			log.Printf("could not find artifacts in release of provider %s version %s, skipping...", p.ProviderName, r.TagName)
 			continue
 		}
-		if (shaSumsArtifact == github.ReleaseAsset{}) {
+		if (versionArtifacts.ShaSumsArtifact == Artifact{}) {
 			return nil, fmt.Errorf("could not SHASUMS artifact for provider %s version %s", p.ProviderName, r.TagName)
 		}
-		if (shaSumsSignatureArtifact == github.ReleaseAsset{}) {
+		if (versionArtifacts.ShaSumsSignatureArtifact == Artifact{}) {
 			return nil, fmt.Errorf("could not SHASUMS signature artifact for provider %s version %s", p.ProviderName, r.TagName)
 		}
 
 		versions = append(versions, provider.Version{
-			Version:             internal.TrimTagPrefix(r.TagName),
-			Protocols:           []string{"5.0"},
-			SHASumsURL:          shaSumsArtifact.DownloadURL,
-			SHASumsSignatureURL: shaSumsSignatureArtifact.DownloadURL,
+			Version:             version,
+			SHASumsURL:          versionArtifacts.ShaSumsArtifact.DownloadURL,
+			SHASumsSignatureURL: versionArtifacts.ShaSumsSignatureArtifact.DownloadURL,
 			Targets:             targets,
 		})
 	}
 
-	// TODO all asset downloads - Shasums and figuring out the protocols
-	//versions, err = enrichWithShaSums(ctx, versions)
-	//if err != nil {
-	//	return nil, err
-	//}
+	versions, err = enrichWithDataFromArtifacts(ctx, versions, versionArtifactsMap)
+	if err != nil {
+		return nil, err
+	}
 
-	return &provider.MetadataFile{
+	mergedMetadata := mergeMetadata(existingMetadata, provider.MetadataFile{
 		Versions: versions,
-	}, nil
+	})
+	return &mergedMetadata, nil
+}
+
+func mergeMetadata(oldMetadata provider.MetadataFile, newMetadata provider.MetadataFile) provider.MetadataFile {
+	versions := append(newMetadata.Versions, oldMetadata.Versions...)
+
+	semverSortFunc := func(a, b provider.Version) int {
+		return semver.Compare(fmt.Sprintf("s%s", a.Version), fmt.Sprintf("s%s", b.Version))
+	}
+	slices.SortFunc(versions, semverSortFunc)
+
+	return provider.MetadataFile{
+		Repository: oldMetadata.Repository,
+		Versions:   versions,
+	}
+}
+
+func getExistingMetadata(p provider.Provider) (provider.MetadataFile, error) {
```

Comment:
For cleaning up after: we should probably have something to read/write metadata that can be shared between this package and the v1 renderer.